### ISSUE TITLE
Regression in ViT Tiny accuracy (classification)

### ISSUE THREAD

**Issue by leoll2** [open-edge-platform/training_extensions#7174]
**Steps to reproduce**

1. Create a classification project in Geti v3.0 importing the [flowers dataset](https://storage.geti.intel.com/test-data/geti/datasets/qa/flowers-geti-medium.zip).
2. Train a ViT-Tiny model
3. Upgrade Geti to v3.1
4. Train again ViT-Tiny (from scratch, NOT from the previous revision)
5. Compare the two models

**Expected**

In v3.1, ViT Tiny achieves basically the same accuracy as it did in v3.0.

**Actual**

On this dataset, ViT Tiny achieves 90% in v3.0 but only 70% in v3.1.

<img width="1871" height="165" alt="Image" src="https://github.com/user-attachments/assets/7a9126ee-b660-4209-86bd-19f8e1268d58" />

***Training metrics***

| **v3.0** | **v3.1** |
|--------|--------|
| <img width="2238" height="790" alt="Image" src="https://github.com/user-attachments/assets/4401568f-6299-403b-ad84-19e59bfffaa5" /> | <img width="2255" height="801" alt="Image" src="https://github.com/user-attachments/assets/1e71ea9b-faf8-4d45-950f-db87e8f7b7f0" /> | 

From the training loss graph, we can observe that the model used to converge much faster.

**Comment by itallix** [comment]
There is a difference in weights used as a base between v3.1 and v3.0:

**v3.1** (app manifest): `https://download.openmmlab.com/mmclassification/v0/deit/deit-tiny_pt-4xb256_in1k_20220218-13b382a0.pth`
**v3.0** (library): `https://storage.googleapis.com/vit_models/augreg/Ti_16-i21k-300ep-lr_0.001-aug_none-wd_0.03-do_0.0-sd_0.0--imagenet2012-steps_20k-lr_0.03-res_224.npz`

Pretrained dataset used in Google AugReg is ImageNet-21k, where OpenMMLab is ImageNet-1k.

From [paper](https://arxiv.org/pdf/2106.10270):

> We recommend to use checkpoints that were pre-trained on more upstream data, and not relying
> only on ImageNet-1k as a proxy for model quality, since ImageNet-1k validation accuracy is inflated
> when pre-training on ImageNet-1k, and more varied upstream data yields more widely applicable
> models

**Comment by leoll2** [comment]
Interesting finding! 

Looking at the [v3.0 manifest](https://github.com/open-edge-platform/geti/blob/bc65d75c1f783aee2a26315863c8df330232df6e/application/backend/app/supported_models/manifests/classification/vit_tiny.yaml#L5), I see that it points to `deit-tiny_pt-4xb256_in1k_20220218-13b382a0.pth` (OpenMMLab weights). Does it mean that this file was downloaded but not used, and that the library was actually loading the AugReg weights?

If so, would it be enough to update the manifest `url` to point to `https://storage.googleapis.com/vit_models/augreg/Ti_16-i21k-300ep-lr_0.001-aug_none-wd_0.03-do_0.0-sd_0.0--imagenet2012-steps_20k-lr_0.03-res_224.npz`?

**Comment by itallix** [comment]
>  Looking at the [v3.0 manifest](https://github.com/open-edge-platform/geti/blob/bc65d75c1f783aee2a26315863c8df330232df6e/application/backend/app/supported_models/manifests/classification/vit_tiny.yaml#L5), I see that it points to deit-tiny_pt-4xb256_in1k_20220218-13b382a0.pth (OpenMMLab weights). Does it mean that this file was downloaded but not used, and that the library was actually loading the AugReg weights?

Yes, exactly. In v3.0, the training ignored the app manifest setting and defaulted to loading the library's pretrained URL (AugReg). Respecting the manifest URL was only implemented starting in v3.1.

> If so, would it be enough to update the manifest url to point to https://storage.googleapis.com/vit_models/augreg/Ti_16-i21k-300ep-lr_0.001-aug_none-wd_0.03-do_0.0-sd_0.0--imagenet2012-steps_20k-lr_0.03-res_224.npz?

Not quite - updating the manifest URL alone isn't enough. We also need to clean up [VisionTransformerWeightsLoader](https://github.com/open-edge-platform/geti/blob/5774079b01c6162f6afc2be74c201dee144d85e4/library/src/getitune/backend/lightning/models/classification/utils/pretrained_weights.py#L116).

Currently, that block assumes incoming weights are in PyTorch `.pth` (DeiT) format, so handing it an `.npz` file will fail. But we can safely remove that conditional section entirely and let the backbone handle weight loading natively.

**Comment by leoll2** [comment]
Alright, let's try to adjust VisionTransformerWeightsLoader to support loading from `.npz` as it did previously.
